src/main/java/seedu/address/logic/LogicManager.java

```diff
@@ -38,6 +42,10 @@ public CommandResult execute(String commandText) throws CommandException, ParseE
         try {
             Command command = addressBookParser.parseCommand(commandText);
             command.setData(model, history, undoRedoStack);
+            if (command instanceof ExportCommand) {
```

Comment:
This makes the `ExportCommand` look like special. It should not be special. Instead, as mentioned above, it should call a method like `exportAddressBook` in model Manager, then the model manager will deal with storage component.

Notice the `setData` method in `Command` abstract class, you can use `this.model` in `ExportCommand` directly.